src/main/scala/com/microsoft/hyperspace/index/IndexCollectionManager.scala

```diff
@@ -66,7 +66,13 @@ class IndexCollectionManager(
     withLogManager(indexName) { logManager =>
       val indexPath = PathResolver(spark.sessionState.conf).getIndexPath(indexName)
       val dataManager = indexDataManagerFactory.create(indexPath)
-      new RefreshAction(spark, logManager, dataManager).run()
+      if (spark.sessionState.conf
```

Comment:
I think a new api parameter might be better than config for this purpose as suggested before. For example,
`hs.refreshIndex(indexName, mode="delete")`

Reply:
I agree, I dont fully recall which suggestion you are referring to - In the discussion that I recall @imback82 suggested to not change the refresh api (for now) and instead protect the delete feature under a Spark config flag so both Append and Delete work can move forward with minimum conflict. API change is supposed to happen as the follow-up when merging them.

Reply:
If you think you guys can move forward without too much conflict, just go ahead. I am fine with either approach.

Reply:
Issue #149 added to clarify the assumptions and the plan for subsequent merge of changes.